src/Data/List/Types.purs

```diff
@@ -91,6 +110,11 @@ instance traversableList :: Traversable List where
   traverse f = map (foldl (flip (:)) Nil) <<< foldl (\acc -> lift2 (flip (:)) acc <<< f) (pure Nil)
   sequence = traverse id
 
+instance traversableWithIndexList :: TraversableWithIndex Int List where
+  traverseWithIndex f =
+    map (foldl (flip (:)) Nil)
```

Comment:
Isn't this `reverse`?

Reply:
Yes, but we can't use it because of the reverse dependency.